tests/utils/index.ts

```diff
@@ -26,7 +28,64 @@ export const ARGS: TestingArguments = {
   cliVersion: process.env.PLUGINS_TEST_CLI_VERSION,
   cliPath: process.env.PLUGINS_TEST_CLI_PATH,
   linterVersion: parseLinterVersion(process.env.PLUGINS_TEST_LINTER_VERSION ?? ""),
+  dumpNewSnapshot: process.env.PLUGINS_TEST_NEW_SNAPSHOT?.toLowerCase() === "true",
 };
-if (ARGS.cliVersion || ARGS.cliPath || ARGS.linterVersion) {
+if (ARGS.cliVersion || ARGS.cliPath || ARGS.linterVersion || ARGS.dumpNewSnapshot) {
   console.debug(ARGS);
 }
+
+/**
+ * Identifies snapshot file to use, based on linter, version, and ARGS.dumpNewSnapshot.
+ *
+ * @param snapshotDirPath absolute path to snapshot directory
+ * @param linterName name of the linter being tested
+ * @param prefix prefix of the file being checked
+ * @param linterVersion version of the linter that was enabled (may be undefined)
+ * @returns absolute path to the relevant snapshot file
+ */
+export const getSnapshotPath = (
+  snapshotDirPath: string,
+  linterName: string,
+  prefix: string,
+  linterVersion?: string
+): string => {
+  // If this is a versionless linter, don't specify a version.
+  if (!linterVersion) {
+    return path.resolve(snapshotDirPath, `${linterName}_${prefix}.shot`);
+  }
+
+  // If this is a versioned linter && dumpNewSnapshot, return its generated name.
+  const specificVersionSnapshotName = path.resolve(
+    snapshotDirPath,
+    `${linterName}_v${linterVersion}_${prefix}.shot`
+  );
+  if (ARGS.dumpNewSnapshot) {
+    return specificVersionSnapshotName;
+  }
```

Comment:
I think that if we want to update the snapshot in question, we can rename `linter_vLatest.out` to `linter_vX.Y.Z.out` and overwrite `linter_vLatest.out`. It seems that right now if the latest version of the linter changes on you, you need to do some manual work _after_ you re-run with `PLUGINS_TEST_NEW_SNAPSHOT=true`?

Reply:
I think there's a slight gap in understanding here. Let's discuss

Reply:
TODO: Use same pattern with formatters and linters

Reply:
TODO: Callout snapshot resolution in readme.